linode/manager
# upcoming: [M3-7951] Update Placement Groups region API limits

## Description 📝
This PR brings CM up to date with the latest region limits ALPHA deployments. This update's only concern is updating the GET region limits API response to match the expected shape.

```TypeScript
export interface Region {  
  id: string;  
  label: string;  
  country: Country;  
  capabilities: Capabilities[];    
  maximum_pgs_per_customer: number;    
  maximum_vms_per_pg: number;    
  status: RegionStatus;  
  resolvers: DNSResolvers;  
  site_type: RegionSite;
} 
```

becomes:

```TypeScript
export interface Region {  
  id: string;  
  label: string;  
  country: Country;  
  capabilities: Capabilities[];  
  placement_group_limits: {    
    maximum_pgs_per_customer: number;    
    maximum_linodes_per_pg: number;  
  };  
  status: RegionStatus;  
  resolvers: DNSResolvers;  
  site_type: RegionSite;
} 
```

It enables the UI to properly handle PG limits. Limits are currently set with hard coded values on the API side, so the behavior is predictable for now:
- `5` max PGs per Customer
- `5` max Linodes per PG

## Changes  🔄
- Update APIv4 Region response
- Update related code 

## Preview 📷
![Screenshot 2024-04-03 at 09 13 07](https://github.com/linode/manager/assets/130582365/52792570-0e00-44c0-be18-8b350c58c1e3)
![Screenshot 2024-04-02 at 14 22 26](https://github.com/linode/manager/assets/130582365/4987116c-65ec-47af-9395-c58265d80d01)
![Screenshot 2024-04-02 at 14 21 48](https://github.com/linode/manager/assets/130582365/289e12d1-6867-419f-9561-13f44cdf7b42)
![Screenshot 2024-04-02 at 14 21 38](https://github.com/linode/manager/assets/130582365/83723fbf-d243-42c5-9b34-fb3e99dc0f90)

## How to test 🧪

⚠️ **Note:** Copy is still WIP (main copy, errors, notices)

### Prerequisites
- Switch your environment to ALHPA and pull code locally
- Login with user `pg-user-1` (1Password)

### Verification steps
- Navigate to http://localhost:3000/placement-groups
- Confirm limits by navigating through and playing with the feature
  - Limit display in tables and summaries
  - Creating placement groups
  - Assigning Linodes
  - Create Linode flow  

## As an Author I have considered 🤔

*Check all that apply*
